dsn.go

```diff
@@ -0,0 +1,31 @@
+package gorm
+
+import (
+	"net/url"
+	"strconv"
+)
+
+type DSN struct {
+	Host    string
+	Port    int
+	User    string
+	Pass    string
+	Db      string
```

Comment:
I think it would be better to call this `DB`

Reply:
Fixed